src/pint/models/stand_alone_psr_binaries/BT_piecewise.py

```diff
@@ -0,0 +1,441 @@
+import astropy.constants as c
+import astropy.units as u
+import numpy as np
+import pint.toa
+from pint import GMsun, Tsun, ls
+from pint.models.stand_alone_psr_binaries.BT_model import BTmodel
+from .binary_generic import PSR_BINARY
+
+
+class BTpiecewise(BTmodel):
+    def __init__(self, axis_store_initial=None, t=None, input_params=None):
+        self.binary_name = "BT_piecewise"
+        super(BTpiecewise, self).__init__()
+        self.axis_store_initial = []
+        self.extended_group_range = []
+        self.d_binarydelay_d_par_funcs = [self.d_BTdelay_d_par]
+        if t is not None:
+            self._t = t
+        if input_params is not None:
+            if self.T0X is None:
+                self.update_input(input_params)
+            elif self.T0X is not None:
+                self.update_input()
+        self.binary_params = list(self.param_default_value.keys())
+        # self.param_aliases.update({"T0X": ["T0X"], "A1X": ["A1X"]})
+
+    def set_param_values(self, valDict=None):
+        # print(valDict)
+        super().set_param_values(valDict=valDict)
+        self.piecewise_parameter_loader(valDict=valDict)
+
+    def piecewise_parameter_loader(self, valDict=None):
+        # print(f"Contents of valDict: {valDict}")
+        self.T0X_arr = []  # initialise arrays to store T0X/A1X values per toa
+        self.A1X_arr = []
+        self.lower_group_edge = (
+            []
+        )  # initialise arrays to store piecewise group boundaries
+        self.upper_group_edge = []
+        piecewise_parameter_information = (
+            []
+        )  # initialise array that will be 5 x n in shape. Where n is the number of pieces required by the model
+        # print(f"valDict:{valDict}")
+        if (
+            valDict is None
+        ):  # If there are no updates passed by binary_instance, sets default value (usually overwritten when reading from parfile)
+            self.T0X_arr = self.T0
+            self.A1X_arr = self.A1
+            self.lower_group_edge = [0]
+            self.upper_group_edge = [1e9]
+            self.piecewise_parameter_information = [
+                0,
+                self.T0,
+                self.A1,
+                0 * u.d,
+                1e9 * u.d,
+            ]
+        else:
+            piece_index = (
+                []
+            )  # iniialise array used to count the number of pieces. Operates by seaching for "A1X_i/T0X_i" and appending i to the array. Can operate if pieces are given out of order.
```

Comment:
Try to avoid comments on the same line as your code, it makes for nasty formatting.